### PR TITLE
Stat: improve color mode migration from singlestat panels

### PR DESCRIPTION
**Pull request by ryantxu**
This is an odd issue reported from:
https://twitter.com/thatcks/status/1402691683121352715

Now that we auto-migrate to the stat panel, this will force everything to use a color/threshold :grimacing: 

This PR will use the primary text color when no color is selected.
image

Also making the line color setting useable:
image

Font sizes are still funky... now sure how/if we should migrate them to fixed sizes

### CODE REVIEW

**Comment by ryantxu** [comment]
I tagged this as 8.0.1 because it will mainly affect people migrating and should help cause fewer surprises

**Comment by zuchka** [comment]
Awesome! working beautifully over here

CleanShot 2021-06-10 at 13 28 00@2x

**Review by zuchka — APPROVED** [review]

File: public/app/plugins/panel/stat/module.tsx [hunk 1]
```diff
@@ -1,11 +1,26 @@
...
 
 export const plugin = new PanelPlugin<StatPanelOptions>(StatPanel)
-  .useFieldConfig()
+  .useFieldConfig({
+    standardOptions: {
+      [FieldConfigProperty.Color]: {
+        settings: {
+          byValueSupport: true,
+          bySeriesSupport: true,
+          preferThresholdsMode: false,
```
Comment (torkelo) [inline public/app/plugins/panel/stat/module.tsx:19]: Why this change? Think it will make it harder to switch to this panel from the graphs. You will always have to change color scheme from palette every time

Comment (torkelo) [reply public/app/plugins/panel/stat/module.tsx:19]: yes, I tested this and this does indeed make stat panels use palette by default (so new stat panels are not yellow)

Comment (ryantxu) [reply public/app/plugins/panel/stat/module.tsx:19]: :(   any ideas how to have the fixed color stick when we add it in migrations?

https://github.com/grafana/grafana/pull/35538/files#diff-8d7549660e15a6f88a67d7c95cae4e98c7602cae18279e227f93e5ffa37841b7R32

I looked everywhere and finally found that without the settings above it would remove the things I just set.

Comment (torkelo) [reply public/app/plugins/panel/stat/module.tsx:19]: Ah missed be bad logic in the change plugin option handling, if the color is set to fixed color the that should not be changed

File: public/app/features/dashboard/state/getPanelOptionsWithDefaults.ts [hunk 2]
```diff
@@ -160,7 +160,7 @@ function adaptFieldColorMode(
 
     // When supporting value colors and prefering thresholds, use Thresholds mode.
     // Otherwise keep current mode
-    if (colorSettings.byValueSupport && colorSettings.preferThresholdsMode) {
+    if (colorSettings.byValueSupport && colorSettings.preferThresholdsMode && mode?.id !== FieldColorModeId.Fixed) {
```
Comment (ryantxu) [inline public/app/features/dashboard/state/getPanelOptionsWithDefaults.ts:163]: @torkelo -- this will always keep fixed when switching panels, is that reasonable?

File: public/app/plugins/panel/stat/module.tsx [hunk 3]
```diff
@@ -34,12 +39,13 @@ export const plugin = new PanelPlugin<StatPanelOptions>(StatPanel)
...
-        defaultValue: 'value',
+        defaultValue: BigValueColorMode.Value,
         category: mainCategory,
         settings: {
           options: [
-            { value: 'value', label: 'Value' },
-            { value: 'background', label: 'Background' },
+            { value: BigValueColorMode.Value, label: 'Value' },
+            { value: BigValueColorMode.Background, label: 'Background' },
+            { value: BigValueColorMode.None, label: 'None' },
```
Comment (torkelo) [inline public/app/plugins/panel/stat/module.tsx:48]: can we move None to first ?

**Review by torkelo — APPROVED** [review]
Looks good!

Judging by other options we usually have None the first
